velox/vector/FlatVector-inl.h

```diff
@@ -425,6 +426,7 @@ void FlatVector<T>::resize(vector_size_t newSize, bool setNotNull) {
       for (auto index = previousSize; index < newSize; ++index) {
         new (&stringViews[index]) StringView();
       }
+      SimpleVector<StringView>::resizeIsAsciiIfNotEmpty(newSize, false);
```

Comment:
If we are going to resize for capacity sake, it shouldnt matter if it is empty right ?

Reply:
If we don't have ascii at the moment and might not need at all, then why allocating memory for it?

Reply:
Sorry maybe there is a misunderstanding, I am just saying that we resize the selectivity vector only if hasSelections() is true . We dont need to check for that and can resize without that . There are many cases where we first create vector of size 0 (via BaseVector::create) and then resize it to something, without adding data. If we want to ensure capacity then we can just resize the ascii selectivity vector without worrying about hasSelections (i.e if not empty ) is what I am saying. Let me know if that is wrong.

Reply:
That's how I understood you in the 1st place.
You basically suggesting resizing unconditionally.
That is not wrong per se.
However:
1. We do not guarantee at the moment that SimpleVector ALWAYS has memory for ascii. That makes sense because we don't need it all the time. So why allocating if we might not need it.
2. But if we have it (calculated ascii for at least 1 row), then we ought to keep ascii memory up to date, in case it is accessed.

Hope this helps.